src/Containers.hs

```diff
@@ -97,9 +98,10 @@ type instance Element BS.ByteString = Word8
 type instance Element BSL.ByteString = Word8
 type instance Element IS.IntSet = Int
 
--- | Type class for container. Fully compatible with 'Foldable'.
+-- | Type class for data types that can be converted to List.
+-- Fully compatible with 'Foldable'.
 -- Contains very small and safe subset of 'Foldable' functions.
-class Container t where
+class ToList t where
```

Comment:
Could you please enhance `ToList` data type in several ways?

1. Add this law for `ToList` type class: `null x ≡ null (toList x)`
2. Add default implementation of `null` function.
3. Describe `{-# MINIMAL #-}` pragma and laws in documentation for this type class.

Check this module as example of such documentation: 

http://hackage.haskell.org/package/base-4.10.0.0/docs/Data-Bifunctor.html